### PR TITLE
INTMDB-638: DataDog ThirdPartyIntegration Documentation update

### PR DESCRIPTION
**Pull request by evertsd**
## Description

Updated third_party_integration documentation to indicate that US3 and US5 are valid regions for DATADOG.
Updated documentation link to latest api reference

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

### CODE REVIEW

**Review by andreaangiolillo — APPROVED** [review]
LGTM

File: website/docs/r/third_party_integration.markdown [hunk 1]
```diff
@@ -57,7 +57,7 @@ Additional values based on Type
   * `service_key` - Your Service Key.
 * `DATADOG`
   * `api_key` - Your API Key.
-  * `region` - Indicates which API URL to use, either US or EU. Datadog will use US by default.    
+  * `region` - Indicates which API URL to use, either US, EU, US3, or US5. Datadog will use US by default.
```
Comment (Zuhairahmed) [inline website/docs/r/third_party_integration.markdown]: @evertsd can you confirm that code changes to this TF resource or to Atlas Go-Client is required to support additional regions?

Comment (Zuhairahmed) [inline website/docs/r/third_party_integration.markdown]:
```suggestion
  * `region` - Indicates which API URL to use, either "US", "EU", "US3", or "US5". Datadog will use "US" by default.    

```

File: website/docs/d/third_party_integrations.markdown [hunk 2]
```diff
@@ -70,7 +70,7 @@ Additional values based on Type
   * `service_key` - Your Service Key.
 * `DATADOG`
   * `api_key` - Your API Key.
-  * `region` - Indicates which API URL to use, either US or EU. Datadog will use US by default.    
+  * `region` - Indicates which API URL to use, either US, EU, US3, or US5. Datadog will use US by default.
```
Comment (Zuhairahmed) [inline website/docs/d/third_party_integrations.markdown]:
```suggestion
  * `region` - Indicates which API URL to use, either "US", "EU", "US3", or "US5". Datadog will use "US" by default.    
```

File: website/docs/d/third_party_integration.markdown [hunk 3]
```diff
@@ -59,7 +59,7 @@ Additional values based on Type
   * `service_key` - Your Service Key.
 * `DATADOG`
   * `api_key` - Your API Key.
-  * `region` - Indicates which API URL to use, either US or EU. Datadog will use US by default.    
+  * `region` - Indicates which API URL to use, either US, EU, US3, or US5. Datadog will use US by default.
```
Comment (Zuhairahmed) [inline website/docs/d/third_party_integration.markdown]:
```suggestion
  * `region` - Indicates which API URL to use, either "US", "EU", "US3", or "US5". Datadog will use "US" by default.    
```

File: examples/third-party-integration/datadog/README.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
+# Example - DataDog third party integration with MongoDB Atlas Cluster
+
+## Dependencies
+
+* Terraform v0.13
+* A [DataDog](https://www.datadoghq.com/) API Key.
+    * As of 2023/03/27, worth noting that this is *not* an Application Key. API Keys can be access via Organization Settings.
```
Comment (Zuhairahmed) [inline examples/third-party-integration/datadog/README.md]:
```suggestion
    * As of March 2023, worth noting that this is *not* an Application Key. API Keys can be created and accessed via Organization Settings from DataDog UI (not from MongoDB Atlas).
```

**Review by Zuhairahmed — APPROVED** [review]
Few comments below, can you also confirm that no code changes are required here or from Atlas Go Client to support these new DataDog regions?

File: website/docs/r/third_party_integration.markdown (hunk 1 above)
Comment (evertsd) [reply website/docs/r/third_party_integration.markdown]: Yep, no code changes required to the tf resource or atlas go-client. I was able to execute an example with the US5 region and have everything go through successfully.

Comment (Zuhairahmed) [reply website/docs/r/third_party_integration.markdown]: nice!

**Comment by evertsd** [comment]
Committed suggestions, thanks @Zuhairahmed

**Pull request merged**